qutip/steadystate.py

```diff
@@ -914,6 +914,92 @@ def _iter_count(r):
         return rhoss
 
 
+def steadystate_floquet(H_0, c_ops, Op_t, w_d=1.0, nmax=3, sparse=False):
+    """
+    Calculates the effective steady state for a driven
+     system with a time-dependent cosinusoidal term:
+
+    .. math::
+
+        \\mathcal{\\hat{H}}(t) = \\hat{H}_0 +
+         \\mathcal{\\hat{O}} \\cos(\\omega_d t)
+
+    Parameters
+    ----------
+    H_0 : :obj:`~Qobj`
+        A Hamiltonian or Liouvillian operator.
+
+    c_ops : list
+        A list of collapse operators.
+
+    Op_t : :obj:`~Qobj`
+        The the interaction operator which is multiplied by the cosine
+
+    w_d : float, default 1.0
+        The frequency of the drive
+
+    nmax : int, default 3
+        The maximum number of iteration for the solver
+
+    sparse : bool, default False
+        Solve for the steady state using sparse algorithms.
+        Actually, dense seems to be faster.
+
+    Returns
+    -------
+    dm : qobj
+        Steady state density matrix.
+    """
+    if sparse:
+        N = H_0.shape[0]
+
+        L_0 = liouvillian(H_0, c_ops).data.tocsc()
+        L_t = liouvillian(Op_t)
+        L_p = (0.5 * L_t).data.tocsc()
+        L_m = L_p  # (0.5 * A_l * L_t).data
+        L_p_array = L_p.todense()
+        L_m_array = L_m.todense()
+
+        Id = sp.eye(N ** 2, format="csc", dtype=np.complex128)
+        S = T = sp.csc_matrix((N ** 2, N ** 2), dtype=np.complex128)
+
+        for n_i in np.arange(nmax, 0, -1):
+            L = sp.csc_matrix(L_0 - 1j * n_i * w_d * Id + L_m.dot(S))
+            L.sort_indices()
+            LU = splu(L)
```

Comment:
Have you tired other choice of solver `splu` vs `spsolve` or `gmres`, etc.
`steadystate` allows the choice of solver and method, while they are not all great, the option exist. 
Does it make sense to have something similar here?

Reply:
I have tried `spsolve` and it is much slower than `splu`, I have tried it for different type of systems. For `gmres`, as far as I know it is only valid for systems of the type `Ax = b` where `b` is a vector. Here `b` is a matrix. 
Thus, I think the only possible options are `splu` (faster) and `spsolve` (slower).